### PR TITLE
fix filtering by list params on API

### PR DESCRIPTION
**Pull request by norkans7**
similar to #34 
@rowanseymour @nicpottier @ericnewcomer please review

### CODE REVIEW

**Comment by rowanseymour** [comment]
Two small additional changes you can make:

1. The read explorer for /messages says *"A full URN to filter messages by (query parameter can be repeated)"* which suggests we meant to use `getlist` but the code is using `get` already so I'd update the read explorer.

2. message labels aren't mentioned in the /messages read explorer.

Finally I just noticed that message labels can contain commas. Searching for messages by label is already using `get` which won't work with a label containing a comma. We should probably limit message labels to word characters and spaces, but since that's already broken that can go in a different PR.

**Comment by norkans7** [comment]
Right I verified and Labels never contains punctuactions https://github.com/nyaruka/rapidpro/blob/master/temba/msgs/models.py#L1355

**Comment by rowanseymour** [comment]
Unless I'm missing something obvious... doesn't look like that `Label.generate_label` method is ever used. LabelCRUDL creates labels directly with whatever punctuation you want.

**Comment by norkans7** [comment]
Oh, yes now I get that

**Comment by norkans7** [comment]
#43 Prevents labels to have name with a comma

**Comment by rowanseymour** [comment]
Looks good. Need this for both chatpro and tracpro

**Comment by rowanseymour** [comment]
Alas back to the drawing board (see hipchat discussion). We're going to keep `getlist` for params that currently use it, and use it for all new params. For consistency we can allow existing params to be specified using either format. For example, use a function like the following to fetch all existing params that currently use just `get`:

```python
def legacy_getlist(request, name, default=None):
    vals = request.QUERY_PARAMS.getlist(name, default)
    if vals and len(vals) == 1:
        return vals[0].split(',')
    else:
        return vals
```

File: temba/api/views.py [hunk 1]
```diff
@@ -735,10 +735,9 @@ def get_queryset(self):
             types = type.split(',')
             queryset = queryset.filter(msg_type__in=types)
 
-        label = self.request.QUERY_PARAMS.get('label', None)
+        label = self.request.QUERY_PARAMS.getlist('label', None)
```
Comment (rowanseymour) [inline temba/api/views.py]: I think this should be a call to `legacy_getlist` to preserve backwards compatibility if someone is using this with labels separated by commas

File: temba/api/views.py [hunk 2]
```diff
@@ -1168,7 +1169,7 @@ def get_queryset(self):
         if name:
             queryset = queryset.filter(name__icontains=name)
 
-        uuids = self.request.QUERY_PARAMS.getlist('uuid', None)
```
Comment (rowanseymour) [inline temba/api/views.py]: Anything currently using `getlist` should be left as is

File: temba/api/views.py [hunk 3]
```diff
@@ -707,42 +704,36 @@ def get_queryset(self):
...
                 queryset = queryset.filter(pk=-1)
 
-        channel = self.request.QUERY_PARAMS.get('channel', None)
-        if channel:
+        channels = self.request.QUERY_PARAMS.getlist('channel', None)
+        if channels:
             try:
-                channel = int(channel)
-                queryset = queryset.filter(channel_id=channel)
+                channels = [int(channel) for channel in channels]
```
Comment (nicpottier) [inline temba/api/views.py]: I don't think we actually need this casting..

**Pull request merged**